### PR TITLE
Rabbitmq with HA based on autocluster plugin with etcd backend.

### PR DESCRIPTION
**Pull request by ss7pro**
This patch introduces rabbitmq in HA. It's based on autocluster plugin with etcd backend (so it introduces etcd as a dependency).

All rabbitmq are missing permament storage. This is intentional as mnesia recovery after a failure requires all instances to be started in exactly reverse order as they were put down.

### CODE REVIEW

File: rabbitmq/docker/Dockerfile.j2 [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# We need at least ver 3.6.6 for this patch https://github.com/rabbitmq/rabbitmq-server/pull/892
+RUN curl -Lo /tmp/rabbitmq-server.deb https://github.com/rabbitmq/rabbitmq-server/releases/download/rabbitmq_v3_6_6_milestone5/rabbitmq-server_3.6.5.905-1_all.deb \
+  && dpkg -i /tmp/rabbitmq-server.deb \
+  && rm -rf /tmp/rabbitmq-server.deb /var/lib/rabbitmq/*
+
+
+COPY fetch-from-github /fetch-from-github
+
+# Build & install rabbit from source
+#RUN /fetch-from-github binarin   rabbitmq-server      37d4fc789d89a1ff981bd8d771013014b4d05147 /tmp/ra \
```
Comment (DTadrzak) [inline rabbitmq/docker/Dockerfile.j2]: Remove commented lines.

File: rabbitmq/manifest.jsonnet [hunk 2]
```diff
@@ -9,7 +9,7 @@ kpm.package({
     name: "stackanetes/rabbitmq",
     expander: "jinja2",
     author: "Quentin Machu",
-    version: "0.1.0",
+    version: "0.0.0",
```
Comment (DTadrzak) [inline rabbitmq/manifest.jsonnet]: Unncessary change.

File: rabbitmq/manifest.jsonnet [hunk 3]
```diff
@@ -19,7 +19,7 @@ kpm.package({
       control_node_label: "openstack-control-plane",
 
       image: {
-        base: "quay.io/stackanetes/stackanetes-%s:barcelona",
+        base: "10.91.96.87:5000/%s:acid",
```
Comment (DTadrzak) [inline rabbitmq/manifest.jsonnet]: Finally Image should be pushed to quay.io.

File: rabbitmq/templates/deployment.yaml.j2 [hunk 4]
```diff
@@ -1,24 +1,20 @@
 apiVersion: extensions/v1beta1
 kind: Deployment
 spec:
-  replicas: 1
+  replicas: 3
```
Comment (DTadrzak) [inline rabbitmq/templates/deployment.yaml.j2]: Maybe we should add new variable in manifest.jsonnet?
fe. misc.replicas ?

File: rabbitmq/templates/deployment.yaml.j2 [hunk 5]
```diff
@@ -1,24 +1,20 @@
...
         app: rabbitmq
     spec:
       nodeSelector:
         {{ deployment.control_node_label }}: enabled
-      # TODO(DTadrzak): it must be removed in future
-      securityContext:
-        runAsUser: 0
       containers:
         - name: rabbitmq
+          command:
```
Comment (DTadrzak) [inline rabbitmq/templates/deployment.yaml.j2]: In order to make it more consistent with other services, image should set kubernetes-entrypoint as a entrypoint by default, then you don't need to pass it via command.

File: rabbitmq/templates/configmaps/rabbitmq-env.conf.yaml.j2 [hunk 6]
```diff
@@ -2,7 +2,13 @@ apiVersion: v1
...
-    RABBITMQ_BOOT_MODULE=rabbit_clusterer
-    RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS="-pa /usr/lib/rabbitmq/lib/rabbitmq_server-3.5.7/plugins/rabbitmq_clusterer-3.5.x-189b3a81.ez/rabbitmq_clusterer-3.5.x-189b3a81/ebin"
-    ERL_EPMD_PORT={{ network.port.epmd }}
+    RABBITMQ_LOGS=-
+    RABBITMQ_SASL_LOGS=-
+    AUTOCLUSTER_TYPE=etcd
+    AUTOCLUSTER_DELAY=15
+    RABBITMQ_USE_LONGNAME=true
+    RABBITMQ_NODENAME=rabbit@{{ network.ip_address }}
+    AUTOCLUSTER_LOG_LEVEL=debug
```
Comment (DTadrzak) [inline rabbitmq/templates/configmaps/rabbitmq-env.conf.yaml.j2]: log level should be configurable or set to war/err

File: etcd/templates/etcd-member-dp.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+              containerPort: 2379
+            - name: etcd-legacy
+              containerPort: 4001
+            - name: etcd-peer
+              containerPort: 2380
+          volumeMounts:
+            - name: varetcd
+              mountPath: /var/etcd
+
+      volumes: {{data_volumes}}
```
Comment (DTadrzak) [inline etcd/templates/etcd-member-dp.yaml]: new line

File: rabbitmq/templates/configmaps/rabbitmq-env.conf.yaml.j2 (hunk 6 above)
Comment (ss7pro) [inline rabbitmq/templates/configmaps/rabbitmq-env.conf.yaml.j2]: Let's keep it on debug level for now.

File: etcd/templates/etcd-member-dp.yaml (hunk 7 above)
Comment (ss7pro) [inline etcd/templates/etcd-member-dp.yaml]: I've taken template from kubespray repo. Would prefer to not change anything here.

**Comment by ss7pro** [comment]
@DTadrzak @PiotrProkop @Quentin-M ready for a review.

**Comment by DTadrzak** [comment]
LGTM

**Pull request merged**